hazelcast/src/main/java/com/hazelcast/map/impl/record/Records.java

```diff
@@ -44,5 +49,18 @@ public static RecordInfo buildRecordInfo(Record record) {
         return info;
     }
 
+    public static Object getValueOrCachedValue(Record record, SerializationService serializationService) {
```

Comment:
I think this method should be defined on record. THat is the most logical place.